### PR TITLE
add nix support

### PR DESCRIPTION
**Pull request by GTrunSec**

### CODE REVIEW

**Comment by BrianHicks** [comment]
Looking good! A couple of questions:

- I recall looking at this grammar before and seeing that it was not completely finished. Did that change? If not, what subset of the language should we say this supports? I ask because so far we've only accepted "finished" grammars.
- could you please add an integration test for some language samples so that we do not regress unnecessarily?

**Comment by GTrunSec** [comment]
> I recall looking at this grammar before and seeing that it was not completely finished. Did that change? If not, what subset of the language should we say this supports? I ask because so far we've only accepted "finished" grammars.

At present, the general feature of the nix-tree-sitter is almost completed. It already meets my current needs, whether it is in emacs for editor or tree-sitter-grep.

**Comment by BrianHicks** [comment]
Sorry, I didn't mean for you to add an entirely new integration target, just a test like those in (I believe) src/main.rs. (Sorry for the brevity; I'm at a company event at the moment and don't have my computer readily available.)

**Comment by GTrunSec** [comment]
Thanks, changed it.

File: flake.nix [hunk 1]
```diff
@@ -81,13 +86,23 @@
...
-          root = ./.;
+          root = inputs.nix-filter.lib.filter {
+            root = ./.;
+              include = [
+                (inputs.nix-filter.lib.inDirectory ./src)
+                ./build.rs
+                ./Cargo.lock
+                ./Cargo.toml
+              ];
+          };
```
Comment (BrianHicks) [inline flake.nix:102]: I'm sorry if I wasn't clear enough here: I don't want to make any changes in this PR aside from adding the new grammar. Could you please clean out all of this?

**Review by BrianHicks — CHANGES_REQUESTED** [review]
thank you for making the changes before! I'm back at my computer now and having a proper look at this.

Aside from the comment I left, would you mind putting all the places that mention Nix into (Latin) alphabetical order instead of adding it to the end of lists? That'd be between Markdown and PHP.

File: flake.nix (hunk 1 above)
Comment (GTrunSec) [reply flake.nix]: Yeh, I could.

**Comment by GTrunSec** [comment]
Thanks, Solved.

**Comment by BrianHicks** [comment]
looks good now, thank you!

**Comment by GTrunSec** [comment]
The nix looks like it had network issues before CI testing.

**Pull request merged**